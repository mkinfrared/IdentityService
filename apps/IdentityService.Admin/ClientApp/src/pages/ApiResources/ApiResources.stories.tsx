import { Meta, StoryObj } from "@storybook/react";

import ApiResources from "./ApiResources";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/ApiResources",
  component: ApiResources,
  tags: ["autodocs"],
} satisfies Meta<typeof ApiResources>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <ApiResources {...args} />
    </div>
  ),
};

export { Default };

export default meta;
