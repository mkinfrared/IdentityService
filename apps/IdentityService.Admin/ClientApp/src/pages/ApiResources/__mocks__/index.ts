export { default } from "./ApiResources.mock";
