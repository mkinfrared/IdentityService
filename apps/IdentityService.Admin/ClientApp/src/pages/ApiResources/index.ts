export { default } from "./ApiResources";

export * from "./ApiResources.type";
