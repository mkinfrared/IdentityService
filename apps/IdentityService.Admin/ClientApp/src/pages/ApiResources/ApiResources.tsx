import css from "./ApiResources.module.scss";
import { ApiResourcesProps } from "./ApiResources.type";

const ApiResources = ({ className }: ApiResourcesProps) => {
  const classNames = [css.ApiResources, className].filter(Boolean).join(" ");

  return (
    <div className={classNames} data-testid="ApiResources">
      ApiResources works!!
    </div>
  );
};

export default ApiResources;
