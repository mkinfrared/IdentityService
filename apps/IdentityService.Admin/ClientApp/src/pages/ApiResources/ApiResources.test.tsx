import { render } from "@testing-library/react";

import ApiResources from "./ApiResources";

describe("<ApiResources />", () => {
  const Component = <ApiResources />;

  it("should be defined", () => {
    expect(ApiResources).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("ApiResources");

    expect(element).toBeDefined();
  });
});
