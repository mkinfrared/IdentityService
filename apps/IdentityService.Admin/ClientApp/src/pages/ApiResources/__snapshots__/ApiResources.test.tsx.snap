// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApiResources /> should match the snapshot 1`] = `
<div>
  <div
    class="ApiResources"
    data-testid="ApiResources"
  >
    ApiResources works!!
  </div>
</div>
`;
